pkg/querytracker/querytracker.go

```diff
@@ -215,6 +215,27 @@ type colUsage struct {
 	usage int
 }
 
+func PopulateColUsageFromMeasure(measureAgg *structs.MeasureAggregator, measureInfoUsage map[string]bool) {
+	if measureAgg == nil {
+		return
+	}
+
+	if measureAgg.ValueColRequest != nil {
+		fields := measureAgg.ValueColRequest.GetFields()
+		for _, field := range fields {
+			measureInfoUsage[field] = true
+		}
+	} else {
+		measureInfoUsage[measureAgg.MeasureCol] = true
+	}
+}
+
+func RemoveIrrelevantCols[T any](cols map[string]T, irrelevantCols []string) {
+	for _, col := range irrelevantCols {
+		delete(cols, col)
+	}
```

Comment:
I think you can make this as a generic function in `maputils.go`.

Reply:
addressed in the latest commit.